src/app/clusters/closure-dimension-server/closure-dimension-cluster-logic.cpp

```
@@ -0,0 +1,469 @@
+    translationDirection = mState.translationDirection;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClusterLogic::GetRotationAxis(RotationAxisEnum & rotationAxis)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mConformance.HasFeature(Feature::kRotation), CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+    rotationAxis = mState.rotationAxis;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClusterLogic::GetOverflow(OverflowEnum & overflow)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError((mConformance.HasFeature(Feature::kRotation) || mConformance.HasFeature(Feature::kMotionLatching)),
+                        CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+    VerifyOrReturnError(mConformance.supportsOverflow, CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+    overflow = mState.overflow;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClusterLogic::GetModulationType(ModulationTypeEnum & modulationType)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mConformance.HasFeature(Feature::kModulation), CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+    modulationType = mState.modulationType;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClusterLogic::GetFeatureMap(Attributes::FeatureMap::TypeInfo::Type & featureMap)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INCORRECT_STATE);
+    featureMap = mConformance.featureMap;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClusterLogic::GetClusterRevision(Attributes::ClusterRevision::TypeInfo::Type & clusterRevision)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INCORRECT_STATE);
+    clusterRevision = kClusterRevision;
+    return CHIP_NO_ERROR;
+}
+
+Status ClusterLogic::HandleSetTargetCommand(Optional<Percent100ths> position, Optional<TargetLatchEnum> latch,
+                                            Optional<Globals::ThreeLevelAutoEnum> speed)
+{
+    VerifyOrReturnError(mInitialized, Status::Failure);
```

I'm not sure if this should return an error or die since the application isn't in a valide state.
I think it should be a VerifyOrDie in this case.